### PR TITLE
Implement database API

### PR DESCRIPTION
**Pull request by jepoorbaugh**
Adds the ability to track information about students starting and completing exercises.

### CODE REVIEW

File: app/blueprints/api.py [hunk 1]
```diff
@@ -0,0 +1,365 @@
...
+    return db
+
+
+def query_db(statement, args=(), one=False):
+    """Executes a SQL statement against the database and returns the results."""
+    cur = get_db().execute(statement, args)
+    results = cur.fetchall()
+    cur.close()
+
+    return (results[0] if results else None) if one else results
```
Comment (jncraton) [inline app/blueprints/api.py:49]: I'm not sure if this functions ultimately simplifies the rest of our code.

File: app/blueprints/api.py [hunk 2]
```diff
@@ -0,0 +1,365 @@
...
+        # 1. Find or Create Section
+        # 1A. Query for section
+        section_query = query_db(
+            """
+            SELECT pk_section_id 
+            FROM section 
+            WHERE txt_section_name = ? 
+                AND txt_instructor_name = ?;
+            """,
+            (request.json["section_name"], request.json["instructor_name"]),
```
Comment (jncraton) [inline app/blueprints/api.py:73]: Using name parameters and the `request.json` dictionary here should simplify most of these requests.

File: app/db/schema.sql [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    txt_student_program_output TEXT,
+    bool_is_complete INTEGER,
+    ts_starting_time INTEGER,
+    ts_submission_time INTEGER,
+    ts_time_recorded INTEGER,
+    fk_exercise_id INTEGER,
+    fk_student_id INTEGER,
+    FOREIGN KEY (fk_exercise_id) REFERENCES exercise (pk_exercise_id)
+    FOREIGN KEY (fk_student_id) REFERENCES student (pk_student_id)
+) STRICT;
```
Comment (jncraton) [inline app/db/schema.sql:38]: Column names should not be prefixed with their type and/or key information.

File: tests/flask_test.py
Comment (jncraton) [inline tests/flask_test.py:1]: It might make sense to use "api" in this filename as we already test flask static file serving.

**Review by jncraton — CHANGES_REQUESTED** [review]

File: app/blueprints/api.py (hunk 2 above)
Comment (jepoorbaugh) [reply app/blueprints/api.py]: This has been addressed in the latest push

File: tests/flask_test.py
Comment (jepoorbaugh) [reply tests/flask_test.py:1]: This has been changed in the latest push

**Review by owenmock22 — CHANGES_REQUESTED** [review]
Remove the shorter event  #listener on the end button in python.js.

**Review by Peyton3995 — APPROVED** [review]

**Review by jncraton — APPROVED** [review]

File: www/index.html [hunk 4]
```diff
@@ -1,4 +1,4 @@
-<!doctype html>
+222<!doctype html>
```
Comment (jncraton) [inline www/index.html:1]: It looks like `222` may have been added here by mistake.

**Review by jncraton — CHANGES_REQUESTED** [review]

**Review by owenmock22 — APPROVED** [review]

**Review by jncraton — APPROVED** [review]

**Pull request merged**